crates/bonsaidb-macros/src/lib.rs

```diff
@@ -528,16 +579,16 @@ pub fn key_derive(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
             # use std::{borrow::Cow, io::{self, ErrorKind}};
             # use #core::key::{ByteSource, KeyVisitor, IncorrectByteLength, Key, KeyEncoding};
 
-            impl #impl_generics Key<'key> for #ident #ty_generics #where_clause {
+            impl #impl_generics Key<$'key> for #ident #ty_generics #where_clause {
                 const CAN_OWN_BYTES: bool = <#ty>::CAN_OWN_BYTES;
 
-                fn from_ord_bytes<'b>(bytes: ByteSource<'key, 'b>) -> Result<Self, Self::Error> {
+                fn from_ord_bytes<$'b>(bytes: ByteSource<$'key, $'b>) -> Result<Self, Self::Error> {
```

Comment:
Had to enable name spacing of the lifetimes due to them clashing with struct lifetimes.